examples/pytorch/ogb/ngnn/seal/main.py

```diff
@@ -0,0 +1,652 @@
+def evaluate_rocauc(y_pred_pos, y_pred_neg):
+    results = {}
+    rocauc = evaluator.eval(
+        {
+            "y_pred_pos": y_pred_pos,
+            "y_pred_neg": y_pred_neg,
+        }
+    )["rocauc"]
+
+    results["rocauc"] = rocauc
+
+    return results
+
+
+def print_log(*x, sep="\n", end="\n", mode="a"):
+    print(*x, sep=sep, end=end)
+    with open(log_file, mode=mode) as f:
+        print(*x, sep=sep, end=end, file=f)
+
+
+if __name__ == "__main__":
+    # Data settings
+    parser = argparse.ArgumentParser(description="OGBL (SEAL)")
+    parser.add_argument("--dataset", type=str, default="ogbl-vessel")
+    # GNN settings
+    parser.add_argument(
+        "--max_z",
+        type=int,
+        default=1000,
+        help="max number of labels as embeddings to look up",
+    )
+    parser.add_argument("--sortpool_k", type=float, default=0.75)
+    parser.add_argument("--num_layers", type=int, default=3)
+    parser.add_argument("--hidden_channels", type=int, default=32)
+    parser.add_argument("--batch_size", type=int, default=32)
+    parser.add_argument(
+        "--ngnn_type",
+        type=str,
+        default="none",
+        choices=["none", "input", "hidden", "output", "all"],
+        help="You can set this value from 'none', 'input', 'hidden' or 'all' to apply NGNN to different GNN layers.",
+    )
+    # Subgraph extraction settings
+    parser.add_argument("--ratio_per_hop", type=float, default=1.0)
+    parser.add_argument(
+        "--use_feature",
+        action="store_true",
+        help="whether to use raw node features as GNN input",
+    )
+    parser.add_argument(
+        "--use_edge_weight",
+        action="store_true",
+        help="whether to consider edge weight in GNN",
+    )
+    # Training settings
+    parser.add_argument(
+        "--device",
+        type=int,
+        default=0,
+        help="GPU device ID. Use -1 for CPU training.",
+    )
+    parser.add_argument("--lr", type=float, default=0.001)
+    parser.add_argument("--epochs", type=int, default=5)
+    parser.add_argument("--dropout", type=float, default=0.0)
+    parser.add_argument("--runs", type=int, default=10)
+    parser.add_argument("--train_percent", type=float, default=5)
+    parser.add_argument("--val_percent", type=float, default=10)
+    parser.add_argument("--final_val_percent", type=float, default=100)
+    parser.add_argument("--test_percent", type=float, default=100)
+    parser.add_argument("--no_test", action="store_true")
+    parser.add_argument(
+        "--dynamic_train",
+        action="store_true",
+        help="dynamically extract enclosing subgraphs on the fly",
+    )
+    parser.add_argument("--dynamic_val", action="store_true")
+    parser.add_argument("--dynamic_test", action="store_true")
+    parser.add_argument(
+        "--num_workers",
+        type=int,
+        default=24,
+        help="number of workers for dynamic dataloaders; using a larger value for dynamic dataloading is recommended",
```

Comment:
I am not sure whether we have a better way to keep it with in 80 chars. Worth to try,